modules/drivers/canbus/can_client/socket/socket_can_client_raw.cc

```diff
@@ -74,7 +72,7 @@ ErrorCode SocketCanClientRaw::Start() {
   // init config and state
   // 1. set receive message_id filter, ie white list
   struct can_filter filter[2048];
-  for (int i = 0; i < 2048; ++i) {
+  for (int i = 0; i < sizeof(filter)/sizeof(filter[0]); ++i) {
```

Comment:
filter at line 74 is fixed to 2048. I don't think it is necessary to modify here. It actually increases the computational complexity within the for loop.

Reply:
The only 'increased computation' is handled by the compiler, after compilation, the result would be identical. There is no difference at run time.

The advantage of this is if some want to change the array size, he/she just need to modify the code in one place when the size is used in many place.

Reply:
It is not reliable to rely on the compiler to optimize. 
we can define a constexpr variable for 2048 here.